src/cmdstan-guide/mcmc_sampling_intro.Rmd

```diff
@@ -98,8 +98,46 @@ One way to monitor whether a chain has converged to the equilibrium distribution
 to other randomly initialized chains.
 For robust diagnostics, we recommend running 4 chains.
 
+There are two different ways of running multiple chains, with the `num_chains` argument using a single executable
+and by using the Unix and DOS shell to run multiple executables. The former is currently supported and 
+recommended when using the NUTS sampling algorithm with either the diagonal (`diag_e`) on dense (`dense_e`) metric.
+
+### Using the num_chains argument to run multiple chains
+
+The `num_chains` argument can be used with the NUTS sampling algorihtm with either the diagonal (`diag_e`) on dense
+(`dense_e`) metric.
+
+Example that will run 4 chains:
+```
+./bernoulli sample num_chains=4 data file=bernoulli.data.json output file=output.csv
+```
+
+If the model was not compiled with `STAN_THREADS=true`, the above command will run 4 chains sequentially and will
+produce the sample in `output_1.csv`, `output_2.csv`, `output_3.csv`, `output_4.csv`. A suffix with the chain id
+is appended to the provided output filename (`output.csv` in the above command).
+
+If the model was compiled with `STAN_THREADS=true`, the chains can run in parallel, with the `num_threads` argument
+defining the maximum number of threads used to run the chains. If the model uses no within-chain parallelization
+(`map_rect` or `reduce_sum` calls), the below command will run 4 chains in parallel, provided there are cores
+available:
+```
+./bernoulli sample num_chains=4 data file=bernoulli.data.json output file=output.csv num_threads=4
+```
+
+If the model uses within-chain parallelization (`map_rect` or `reduce_sum` calls), the threads are automatically
+scheduled to run the parallel parts of a single chain or run the sequential parts of another chains. The below
+call starts 4 chains that can use 16 threads. At a given moment a single chain may use all 16 threads,
+1 thread, anything in between, or can wait for a thread to be available. The scheduling is left to the Threading
+Building Blocks scheduler.
```

Comment:
Is there some relevant part of the TBB manual we can link to here?

Reply:
The TBB doc was not super stable with their URLs in the past and most users won't need the tech details, I think. Instead, we could link to the wikipedia page maybe:

https://en.wikipedia.org/wiki/Threading_Building_Blocks

?

Reply:
Maybe https://software.intel.com/content/www/us/en/develop/documentation/tbb-documentation/top/intel-threading-building-blocks-developer-guide/the-task-scheduler/how-task-scheduling-works.html ?

Reply:
It was genuinely a question - if there's no good place to link, I'm happy with how it is now. If we think the link @rok-cesnovar posted is stable, I think it would be great to include - but only if it's not going to break super soon.

Reply:
No idea when they break... but I have seen TBB links get broken while working on the project...

Reply:
We could still link to it and fix it if it breaks and somebody notices. The fact that there is a version number and updated date on the page gives me some confidence that it will be semi-stable. 

Or not. I'm fine with whichever @rok-cesnovar wants to do

Reply:
I added the link. It looks stable enough to where I am somewhat certain it will be there. Made a note in the release checklist to check it next time just to be safe.